src/analysis/FVA/fluxVariability.m

```diff
@@ -255,7 +255,12 @@
 else
     sol = solveCobraLP(QuickProblem);
 end
+if sol.stat~=1
+    error('Maximization of the sum of all fluxes failed, even though the first FBA was optimal. Possible cause of error: unbounded fluxes');
+end
+    

```

Comment:
Hmmm... We don't really need to fail at all if we have unbounded fluxes, or infinite bounds.
Essentially, these fluxes just have to be excluded from the initial problem and handled specially in the individual flux checks. i.e. `fluxVariability` would return infinite if the solution for a flux with infinite bounds is unbounded. 

The only problem is if `Vmin` and `Vmax` are requested. The question is what we return then. I personally would return a `NaN` vector.

Reply:
You're right, I took the easy way out. I'll try to implement your suggestion.

Reply:
👍

Reply:
I am finding a couple of bugs, like:

An obvious one (typo, should be LPproblem): 
https://github.com/opencobra/cobratoolbox/blob/7b92e46b476cfaf1cd377cc8099b759ce222dd57/src/analysis/FVA/fluxVariability.m#L391

And this I understand is a bug (ignores even if argin is method='fba'): 
https://github.com/opencobra/cobratoolbox/blob/7b92e46b476cfaf1cd377cc8099b759ce222dd57/src/analysis/FVA/fluxVariability.m#L117-L121

How should I proceed ?

Reply:
The first one: good for spotting that. Just fix it. 

The second one: That is not a bug. 
If neither Vmin nor Vmax are requested (i.e. if nargout < 3), there is no point in calculating them, except for wasting time. Method will never change the actual min/max values. Only the solution that corresponds to it.

Reply:
ok, I was just confused by minNorm being the name for pointing that Vmin/Vmax were required, instead of: 
https://github.com/opencobra/cobratoolbox/blob/7b92e46b476cfaf1cd377cc8099b759ce222dd57/src/analysis/FVA/fluxVariability.m#L104-L108

Which is actually unused.

Reply:
Err well... 
This function is around now for a long time and has had multiple mergers.. 
Looking at it:
could you replace the 'OutputMatrix' variables in that statement, by `minNorm` and remove these lines:

https://github.com/opencobra/cobratoolbox/blob/7b92e46b476cfaf1cd377cc8099b759ce222dd57/src/analysis/FVA/fluxVariability.m#L117-L121

as they are in an entirely unrelated if clause.... (which I assume is due to some file merge).

Reply:
Ok, but just because you ask me :) I actually think that outputMatrix better reflects what the variable is supposed to flag.